pr_agent/tools/ticket_pr_compliance_check.py

```diff
@@ -32,27 +37,26 @@ def extract_ticket_links_from_pr_description(pr_description, repo_path):
     """
     Extract all ticket links from PR description
     """
-    github_tickets = []
+    github_tickets = set()
     try:
-        # example link to search for: https://github.com/Codium-ai/pr-agent-pro/issues/525
-        pattern = r'https://github[^/]+/[^/]+/[^/]+/issues/\d+' # should support also github server (for example 'https://github.company.ai/Codium-ai/pr-agent-pro/issues/525')
-
-        # Find all matches in the text
-        github_tickets = re.findall(pattern, pr_description)
-
-        # Find all issues referenced like #123 and add them as https://github.com/{repo_path}/issues/{issue_number}
-        issue_number_pattern = r'#\d+'
-        issue_numbers = re.findall(issue_number_pattern, pr_description)
-        for issue_number in issue_numbers:
-            issue_number = issue_number[1:]  # remove #
-            # check if issue_number is a valid number and len(issue_number) < 5
-            if issue_number.isdigit() and len(issue_number) < 5:
-                github_tickets.append(f'https://github.com/{repo_path}/issues/{issue_number}')
+        # Use the updated pattern to find matches
+        matches = GITHUB_TICKET_PATTERN.findall(pr_description)
+
+        for match in matches:
+            if match[0]:  # Full URL match
+                github_tickets.add(match[0])
+            elif match[1]:  # Shorthand notation match: owner/repo#issue_number
+                owner, repo, issue_number = match[2], match[3], match[4]
+                github_tickets.add(f'https://github.com/{owner}/{repo}/issues/{issue_number}')
+            else:  # #123 format
+                issue_number = match[5][1:]  # remove #
+                if issue_number.isdigit() and len(issue_number) < 5:
+                    github_tickets.add(f'https://github.com/{repo_path}/issues/{issue_number}')
```

Comment:
**Suggestion:** Add input validation to prevent potential runtime errors caused by invalid input [Possible issue, importance: 8]
```suggestion
                if issue_number.isdigit() and len(issue_number) < 5 and repo_path:
                    github_tickets.add(f'https://github.com/{repo_path}/issues/{issue_number}')
```